weblab/entities/views.py

```diff
@@ -685,8 +689,8 @@ def post(self, request, *args, **kwargs):
             form.add_error(None, 'No changes were made for this version')
             return self.form_invalid(form)
 
-    def fail_with_git_errors(self, git_errors):
-        form = self.get_form()
+    def fail_with_git_errors(self, git_errors, form):
```

Comment:
This method is called in another place too which doesn't pass the `form` argument - interestingly because we don't see the invalid tag error any more! We might be able to trigger it if we add a test with a duplicate (but otherwise valid) tag added to a new version.

Also 4 lines up we have another `form = self.get_form()` line that should be removed.

Reply:
Thanks Jonathan, sorry I'm not quite clear about the first part of this, where you say we dont see the invalid tag error any more. I thought I was checking for this in the rendered_content?

Reply:
It's on [line 665](https://github.com/ModellingWebLab/WebLab/pull/310/files#diff-4bef1f458b6904ff9e9b2d64655045de1bc3c755be9d9f7968a972239941b85aL665) in that file, where a git command line error adding the tag is caught. Since we check for invalid tags in the form, we're no longer getting errors when calling git!